File: Python3/1001 - 1200/1033 Moving Stones Until Consecutive.py
# 24ms, 90.78%; 12.7MB, 100.00%
class Solution:
    def numMovesStones(self, a: int, b: int, c: int) -> list:
        a, b, c = sorted([a, b, c])
        diff1, diff2, diff3 = b - a, c - b, c - a - 2
        left = (0 if diff3 == 0 else 1 if diff1 < 3 or diff2 < 3 else 2)
        return [left, diff3]

File: 1001 - 1200/1033 Moving Stones Until Consecutive.cc
// 0ms, 100.00%; 8.3MB, 90.00%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> numMovesStones(int a, int b, int c) {
        if (a > b) swap(a, b);
        if (a > c) swap(a, c);
        if (b > c) swap(b, c);
        int diff1 = b - a, diff2 = c - b, diff3 = c - a - 2;
        int left = (diff3 == 0 ? 0 : diff1 < 3 || diff2 < 3 ? 1 : 2);
        return vector<int>{left, diff3};
    }
};
